### ISSUE TITLE
Audio is not defined.

### ISSUE THREAD

**Issue by gabrielpgava** [yudielcurbelo/react-qr-scanner#26]
So, i am trying to run the package on a next JS app, and i get this error, some on know how to fix?

<img width="1008" alt="Captura de Tela 2024-02-20 às 22 14 57" src="https://github.com/yudielcurbelo/react-qr-scanner/assets/11247586/998b058d-76a0-4f5a-86a0-29c69a70b26b">

**Comment by yudielcurbelo** [comment]
What version are you using?

**Comment by gabrielpgava** [comment]
I am using this versions     
    "@yudiel/react-qr-scanner": "^1.2.10",
    "next": "^14.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"

**Comment by yudielcurbelo** [comment]
That error is from a previous version. In `1.2.10` it was fixed. Could you check that your .lock file is referencing that version? You could also try to delete your node_modules and recreate it. Please let me know.

**Comment by gabrielpgava** [comment]
The `. lock` file is referencing version `1.2.10`, and I tried to delete the node_modules, but I had no success in solving the issue.

**Comment by yudielcurbelo** [comment]
I have multiple apps deploy to vercel and none of them is presenting this issue with the latest version. I also created a [Sandbox](https://codesandbox.io/p/sandbox/react-qr-scanner-rdpxlc) and was not able to reproduce it

**Comment by gabrielpgava** [comment]
Thanks for your help! I'll keep looking for a solution and let you know when I find it.

**Comment by nick-tw** [comment]
you probably have SSR on the component. make sure it's rendered on the client

```javascript
import dynamic from 'next/dynamic'

const DynamicQrScanner = dynamic(() => import('@yudiel/react-qr-scanner').then(mod => mod.QrScanner), {
    ssr: false
  });

<DynamicQrScanner
                    containerStyle={{
                        height:"100%",
                        paddingTop: "unset"
                    }}
                    onDecode={(result) => console.log(result)}
                    onError={(error) => console.log(error?.message)}
                    />
```

**Comment by Aerofolio** [comment]
I was having the same issue and @nick-tw answer solved my problem. Thank you so much.

**Comment by yudielcurbelo** [comment]
This should be fixed in the beta version `2.0.0-beta.2`. Please note that there are also many changes in this version. Go over the README.

**Comment by walle233** [comment]
I have the same problem

version: ^2.0.0-beta.3

**Comment by NadirBakhsh** [comment]
I am Also facing the same problem.

but I resolved  it.

the problem is that Audio URL is underfed on server side. node is server endearment, so it not supported Audio JavaScript API.

Indicational state is false

![image](https://github.com/yudielcurbelo/react-qr-scanner/assets/46098938/631bcc6f-68f4-4b80-a87f-2363cc111847)

![image](https://github.com/yudielcurbelo/react-qr-scanner/assets/46098938/a97cf8ef-6e3a-49de-94ad-61635a80d184)

![image](https://github.com/yudielcurbelo/react-qr-scanner/assets/46098938/66a56cf5-d826-47fe-9cf3-64c83d189dce)

i hope it will work for everyone

 
![image](https://github.com/yudielcurbelo/react-qr-scanner/assets/46098938/0348dded-6435-43ac-a881-b31085473cf5)

**Comment by yudielcurbelo** [comment]
This issue should be fixed on `2.0.0-beta.8`. There are breaking changes for this beta release.
